tests/e2e/support/page/files/publicLink.ts

```diff
@@ -0,0 +1,229 @@
+import { Actor } from '../../types'
+import { filesCta } from '../../cta'
+import { Locator } from '@playwright/test'
+import util = require('util')
+
+export class PublicLink {
+  private readonly actor: Actor
+  private readonly publicLinkButtonLocator: Locator
+  private readonly publicLinkPasswordLocator: Locator
+  private readonly publicLinkNameLocator: Locator
+  private readonly roleDropdownLocator: Locator
+  private readonly expirationDateDropdownLocator: Locator
+  private readonly createLinkButtonLocator: Locator
+  private readonly publicLinkLocator: Locator
+  private readonly yearButtonLocator: Locator
+  private readonly nextSpanYearLocator: Locator
+  private readonly monthAndYearDropdownLocator: Locator
+  private readonly daySelector: string
+  private readonly monthSelector: string
+  private readonly yearSelector: string
+  private readonly publicLinkListSelector: string
+  private readonly roleSelector: string
+  private readonly folderSelector: string
+
+  private dateType: string
+  constructor({ actor }: { actor: Actor }) {
+    this.actor = actor
+    this.publicLinkButtonLocator = this.actor.page.locator('#files-file-link-add')
+    this.publicLinkPasswordLocator = this.actor.page.locator('#oc-files-file-link-password')
+    this.publicLinkNameLocator = this.actor.page.locator('#oc-files-file-link-name')
+    this.roleDropdownLocator = this.actor.page.locator('#files-file-link-role-button')
+    this.expirationDateDropdownLocator = this.actor.page.locator('#files-links-expiration-btn')
+    this.createLinkButtonLocator = this.actor.page.locator('#oc-files-file-link-create')
+    this.publicLinkLocator = this.actor.page.locator(
+      '//a[@class = "oc-files-file-link-url oc-text-truncate"]'
+    )
+    this.yearButtonLocator = this.actor.page.locator(
+      `//div[@class = "vc-nav-container"]/div[@class="vc-nav-header"]//span[position()=2]`
+    )
+    this.nextSpanYearLocator = this.actor.page.locator(
+      `//div[@class = "vc-nav-container"]/div[@class="vc-nav-header"]//span[position()=3]`
+    )
+    this.monthAndYearDropdownLocator = this.actor.page.locator(`//div[@class = 'vc-title']`)
+    this.daySelector = `//span[@tabindex='-1' or @tabindex='0'][@aria-label='%s']`
```

Comment:
You can add a comment here on why we are using `this.actor.page.locator` on other selectors but not here